### PR TITLE
sync with omicron/r10 nexus.json

### PR DESCRIPTION
**Pull request by ahl**
This is not ready to be merged.

@Nieuwejaar this may step on your toes in #808 and my apologies if so

@rcgoodfellow I have a couple of areas I'm not sure how to fill in; I'll note them in the code

### CODE REVIEW

File: cli/src/cli_builder.rs [hunk 1]
```diff
@@ -479,6 +479,10 @@ fn xxx<'a>(command: CliCommand) -> Option<&'a str> {
             Some("system networking bgp announce-set delete")
         }
         CliCommand::NetworkingBgpAnnounceSetList => Some("system networking bgp announce-set list"),
+        CliCommand::NetworkingBgpAnnouncementList => {
+            Some("system networking bgp announcement list")
+        }
+        CliCommand::NetworkingBgpExported => Some("system networking bgp exported"),
```
Comment (ahl) [inline cli/src/cli_builder.rs:485]: @rcgoodfellow and @Nieuwejaar is this what we want here?

File: cli/src/cmd_net.rs [hunk 2]
```diff
@@ -262,8 +266,8 @@ impl AuthenticatedCmd for CmdBgpAnnounce {
             .into_inner()
             .into_iter()
             .map(|x| BgpAnnouncementCreate {
-                address_lot_block: NameOrId::Id(x.address_lot_block_id),
-                network: x.network,
+                address_lot_block: todo!(),
+                network: todo!(),
```
Comment (ahl) [inline cli/src/cmd_net.rs]: @rcgoodfellow what should this be?

File: cli/src/cmd_net.rs [hunk 3]
```diff
@@ -314,8 +318,8 @@ impl AuthenticatedCmd for CmdBgpWithdraw {
...
```
Comment (ahl) [inline cli/src/cmd_net.rs]: @rcgoodfellow ditto

File: cli/tests/test_net.rs [hunk 4]
```diff
@@ -167,7 +167,7 @@ fn test_port_config() {
             autoneg: false,
             fec: LinkFec::None,
             link_name: String::from("phy0"),
-            lldp_service_config_id: Uuid::new_v4(),
+            lldp_link_config_id: None,
```
Comment (ahl) [inline cli/tests/test_net.rs:170]: @Nieuwejaar this seems to differ from what you have in #808--it looks like you had a required field

**Review by ahl — COMMENTED** [review]
@Nieuwejaar and @rcgoodfellow please feel empowered to push to this branch -- no need to go through code review to get where we need

**Comment by Nieuwejaar** [comment]
> This is not ready to be merged.
> 
> @Nieuwejaar this may step on your toes in #808 and my apologies if so

It looks like this includes and/or subsumes my PR, which is totally cool with me.

File: cli/src/cli_builder.rs (hunk 1 above)
Comment (internet-diglett) [reply cli/src/cli_builder.rs:485]: For the first one, that looks good and is what I tested with when making the updates in Omicron [here](https://github.com/oxidecomputer/omicron/pull/6362#issuecomment-2297504132)

Comment (rcgoodfellow) [reply cli/src/cli_builder.rs:485]: yes this is good

File: cli/src/cmd_net.rs (hunk 2 above)
Comment (rcgoodfellow) [reply cli/src/cmd_net.rs]: fixed

File: cli/src/cmd_net.rs (hunk 3 above)
Comment (rcgoodfellow) [reply cli/src/cmd_net.rs]: fixed

File: cli/tests/test_net.rs (hunk 4 above)
Comment (Nieuwejaar) [reply cli/tests/test_net.rs:170]: You're right.  This is fine.  Even if the LLDP stuff is configured, the `system networking switch-port-settings show` doesn't include it in the output, so it doesn't matter whether this data structure is set up or not.

**Pull request merged**